File: Cpp/771_Jewels_and_Stones.cpp
class Solution {
public:
    int numJewelsInStones(string J, string S) {
        int result = 0;
        unordered_map<char, int> frequency;
        for(auto& ch : S){
            ++frequency[ch];
        }
        for(auto& ch : J){
            result += frequency[ch];
        }
        
        return result;
    }
};

File: Kotlin/771_Jewels_and_Stones.kt
class Solution {
    fun numJewelsInStones(jewels: String, stones: String): Int {
        var result = 0
        val frequency = IntArray(128) {0}
        
        jewels.forEach {
            frequency[it.toInt()] += 1
        }
        stones.forEach {
            result += frequency[it.toInt()]
        }
        
        return result
    }
}